flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/Observer.java

```diff
@@ -129,13 +135,53 @@ private boolean observeFlinkJobStatus(FlinkDeployment flinkApp, Configuration ef
         }
     }
 
+    public boolean observeSavepointStatus(FlinkDeployment flinkApp, Configuration effectiveConfig) {
+        SavepointInfo savepointInfo = flinkApp.getStatus().getJobStatus().getSavepointInfo();
+        if (savepointInfo.getTriggerId() == null) {
+            LOG.debug("Checkpointing not in progress");
+            return true;
+        }
+        SavepointFetchResult savepointFetchResult;
+        try {
+            savepointFetchResult = flinkService.fetchSavepointInfo(flinkApp, effectiveConfig);
+        } catch (Exception e) {
+            LOG.error("Exception while fetching savepoint info", e);
+            return false;
+        }
+
+        if (!savepointFetchResult.isTriggered()) {
+            String error = savepointFetchResult.getError();
+            if (error != null
+                    || !SavepointUtils.withinGracePeriod(operatorConfiguration, savepointInfo)) {
+                LOG.error("Savepoint not triggered, setting reconciliation error");
```

Comment:
Also log the error if not null

Reply:
That is a failure case, I'll adjust the code to make it more clear

Reply:
The webhook is depend on the operator module now, we can let the webhook to use the util in the operator